### PR TITLE
Fix Sketch Palettes for Sketch 53

### PR DESCRIPTION
**Pull request by kristof**
#77  – I'm able to load, save and clear palettes again **ONLY** for document colors.
Sketch appears to have removed `app.globalColors`, still figuring out the proper replacement for this.

But at least some if it is working again.

**EDIT** – Global assets/colors is also fixed.

### CODE REVIEW

**Comment by kristof** [comment]
And global colors + gradients is also fixed.

File: Sketch Palettes.sketchplugin/Contents/Sketch/sketchPalettes.js [hunk 1]
```diff
@@ -45,130 +45,130 @@ function savePalette(context) {
...
 		return;
 	}
-	
+
 	// Get Presets from selected section
 	if (selectSource.indexOfSelectedItem() == 0) {
 		var assets = doc.documentData().assets();
 	} else if (selectSource.indexOfSelectedItem() == 1) {
-		var assets = app.globalAssets();
+		MSPersistentAssetCollection.sharedGlobalAssets();
```
Comment (mathieudutour) [inline Sketch Palettes.sketchplugin/Contents/Sketch/sketchPalettes.js]:
```suggestion
		var assets = MSPersistentAssetCollection.sharedGlobalAssets();
```

File: Sketch Palettes.sketchplugin/Contents/Sketch/sketchPalettes.js [hunk 2]
```diff
@@ -196,60 +196,61 @@ function loadPalette(context) {
...
 			for (var i = 0; i < colorPalette.length; i++) {
-				colors.push(MSColor.colorWithRed_green_blue_alpha(
+				var mscolor = MSColor.colorWithRed_green_blue_alpha(
 					colorPalette[i].red,
 					colorPalette[i].green,
 					colorPalette[i].blue,
 					colorPalette[i].alpha
-				));	
+				);
+				colorAssets.push(MSColorAsset.alloc().initWithAsset_name(mscolor, ''));
```
Comment (mathieudutour) [inline Sketch Palettes.sketchplugin/Contents/Sketch/sketchPalettes.js]:
```suggestion
				colorAssets.push(MSColorAsset.alloc().initWithAsset_name(mscolor, null));
```

File: Sketch Palettes.sketchplugin/Contents/Sketch/sketchPalettes.js [hunk 3]
```diff
@@ -278,22 +279,22 @@ function loadPalette(context) {
 				msgradient.setFrom({ x: fromValue[0], y: fromValue[1] });
 				msgradient.setTo({ x: toValue[0], y: toValue[1] });
 
-				gradients.push(msgradient);
-				
+				gradientAssets.push(MSGradientAsset.alloc().initWithAsset_name(msgradient, ''));
```
Comment (mathieudutour) [inline Sketch Palettes.sketchplugin/Contents/Sketch/sketchPalettes.js]:
```suggestion
				gradientAssets.push(MSGradientAsset.alloc().initWithAsset_name(msgradient, null));
```

**Review by mathieudutour — CHANGES_REQUESTED** [review]

**Comment by kristof** [comment]
@andrewfiorillo Let me know if any edits are needed + how you want to bump the version number.

**Comment by kristof** [comment]
@mathieudutour thanks for the suggestions and a moment of your valuable time!

**Comment by andrewfiorillo** [comment]
@kristof  @mathieudutour thanks for all of your help getting this fixed up! Really appreciate it.

**Comment by himynameisubik** [comment]
Amazing guys @kristof @mathieudutour @andrewfiorillo , thanks a lot for fixing this so fast!

**Pull request merged**